### PR TITLE
Task #58548 [LiconKA] Function-7 : Mix 2 or n Audio Files (part 1)

### PR DESCRIPTION
**Pull request by liconrepo**

### CODE REVIEW

File: app/src/main/java/com/training/soundstudio/ui/fragment/RecordListFragment.java [hunk 1]
```diff
@@ -178,4 +178,15 @@ public void notifyNewFile(String filePath) {
         mRecordListAdapter.notifyDataSetChanged();
         showContent();
     }
+
+    public RecyclerView getRecyclerView() {
+        return mRecyclerView;
+    }
+
+    public int getItemCount() {
+        if (mRecordItemsPaginated.size() > 0) {
```
Comment (b1acKr0se) [inline app/src/main/java/com/training/soundstudio/ui/fragment/RecordListFragment.java]: totally unnecessary

File: app/src/main/java/com/training/soundstudio/util/FFMPEGUtil.java [hunk 2]
```diff
@@ -36,6 +39,20 @@ public static String getCutCommand(String inputFileUrl, String outputFileUrl,
         return stringBuilder.toString();
     }
 
+    public static String getConvertCommand(String inputFileUrl, String outputFileUrl) {
+        if ((TextUtils.isEmpty(inputFileUrl) && (TextUtils.isEmpty(outputFileUrl)))) {
```
Comment (b1acKr0se) [inline app/src/main/java/com/training/soundstudio/util/FFMPEGUtil.java]: What will happen if one of the parameters is null?

File: app/src/main/java/com/training/soundstudio/util/FFMPEGUtil.java [hunk 3]
```diff
@@ -80,4 +98,86 @@ public void onFinish() {
...
+    -map '[out]' output.wav
+
+    #2
+    ffmpeg -i 'concat:input1|input2' -codec copy output
+
+    #3
+    ffmpeg -i input1.mp3 -i input2.mp3 -i input3.mp3 -filter_complex concat=n=3:v=0:a=1 -f MOV -vn -y input.m4a*/
+
+    public static String getConcateCommand(ArrayList<String> inputFileUrls, String outputFileUrl) {
+        if ((inputFileUrls.size() < 0) && (TextUtils.isEmpty(outputFileUrl))) {
```
Comment (b1acKr0se) [inline app/src/main/java/com/training/soundstudio/util/FFMPEGUtil.java]: Again, would the AND operator work here?

File: app/src/main/java/com/training/soundstudio/util/FileUtil.java [hunk 4]
```diff
@@ -216,4 +261,103 @@ public static void shareFileUsingIntent(File file, Activity activity) {
...
+            }
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+
+        List<Track> audioTracks = new LinkedList<Track>();
+
+        for (Movie m : inMovies) {
+            for (Track t : m.getTracks()) {
+                if (t.getHandler().equals("soun")) {
```
Comment (b1acKr0se) [inline app/src/main/java/com/training/soundstudio/util/FileUtil.java]: What does this do?

File: app/src/main/java/com/training/soundstudio/util/FileUtil.java [hunk 5]
```diff
@@ -216,4 +261,103 @@ public static void shareFileUsingIntent(File file, Activity activity) {
...
+            }
+            if (destination != null) {
+                destination.close();
+            }
+        }
+    }
+
+    public static void deleteCachedFile(File cachedFile) {
+        if (cachedFile.exists()) {
+            if (cachedFile.delete()) ;
```
Comment (b1acKr0se) [inline app/src/main/java/com/training/soundstudio/util/FileUtil.java]: Is it okay to leave it like this? What's the point of the `if` when there is no statement inside?

**Comment by b1acKr0se** [comment]
OK

**Comment by nvmanh** [comment]
OK!

**Pull request merged**